week2/index.js

```diff
@@ -0,0 +1,213 @@
+        });
+    });
+}
+
+//add the new item to the array
+function addTodoItem(todoItem){
+    if(todoItem !== ''){
+        if(todoItem.trim() !== ''){
+            todos.push(todoItem.trim());
+            todosWriter(todos);
+        }
+    } else {
+        console.log('Not the correct usage! Try "node index.js help"');        
+    }
+}
+
+// re-write existing file with new todos array 
+function todosWriter(itemsArr){
+    itemsArr.filter((a) => a !== '');
+    fs.writeFile('./todos.txt', itemsArr.join('\r\n'), 'UTF-8', (err) => {
+        if(err) console.log(err);
+    });
+}
+
+//remove item based on filled index
+function itemRemover(index){
+    //if index was actually filled in
+    if(index !== undefined){
+        index = parseInt(index) - 1;
+        fs.readFile('./todos.txt', 'UTF-8', (err, data) => {
+            data = data.split('\r\n');
+            //if that index actually existed in the todo list
+            if(data[index] !== undefined){
+                data.splice(index, 1);
+                todosWriter(data);
+                return;
+            }
+            console.log('You don\'t have that much items on your list!');
+        });
+    } else {
+        console.log('Not the correct usage! Try "node index.js help"');
+    }
+}
+
+//  empty the todos file
+function itemsResetter(){
+    fs.writeFile('./todos.txt', '', (err) => {
+        if (err) console.log(err);
+    });
+    console.log('Resetted the todo list');
+}
+
+
+//  modify an item based on filled index
+function update(index){
+    //check if an index was actually filled in
+    if(index !== undefined){
+        index = parseInt(index) - 1;
+        let rl = readline.createInterface(process.stdin, process.stdout);
+        fs.readFile('./todos.txt', 'UTF-8', (err, data) => {
+            data = data.split('\r\n');
+            rl.question(`the item to be modified is"${data[index]}", what will it be? `, (answer) => {
+                data[index] = answer;
+                todosWriter(data);
+                rl.close();
+            });
+        });
+    } else {
+        console.log('Not the correct usage! Try "node index.js help"');                
+    }
+    
+}
+
+
+//  mark item as DONE
+function crossItem(index){
+    //  Check if an index was actually filled in
+    if(index !== undefined){
+        fs.readFile('./todos.txt', 'UTF-8', (err, data) => {
+            data = data.split('\r\n');
+            index = parseInt(index) - 1;  
+            //check if there's a corresponding todo item for the index and if the item doesn't have the word 'DONE' in it          
+            if(data[index] !== undefined && data[index].indexOf('DONE') === -1){
+                let item = data[index];
+                data[index] += '\t\t\tDONE';//append the word 'DONE'
+                console.log('crossed!');
+                todosWriter(data);
+                return;
+            } 
+            //if the todo item already has the word 'DONE' in it
```

Comment:
Interesting technique... I think it works, but is probably not the most sustainable. What if we want to start adding other fields of information other than 'Done', like 'Assignee' and 'Priority'? (see the main comment about JSON)

Reply:
Yes I don't fully understand how I'm going to make what you're talking about work with JSON but I'll try my best to figure it out.

Reply:
Thanks a lot for you feedback!!

Reply:
Right now, your file looks like this:
```
TodoItem line one.
Another item on line two.
```

You could alternatively use a JSON schema, something like:
```
{
    todos: [
        {
            text: "TodoItem line one.",
            done: true
         },
         {
             text: "Another item on line two.",
             done: false
         }
    ]
}
```

That allows you to store arbitrary data (meaning whatever kinds of information you want) about each todo item.

Reply:
Hi Jason, thanks now I get it.
Actually I already improvised something, I don't know if you think it's good or not but it adds more flexibility to the 'cross' thing, now it's called 'mark' hahaha.
Please let me know what you think of it if you had time. Thanks!